### PR TITLE
fix downloader sub-directories missing

### LINKED ISSUE #119: The document downloader fails when the manifest lists sub-directories
If the manifest contains files in sub-directories, the document downloader [doesn't create](https://github.com/w3c/echidna/blob/master/lib/document-downloader.js#L67) the sub-directory before downloading the resource. It ends up failing with the following error:
```
...
"jobs": {
    "retrieve-resources": 
{
    "status": "error",
    "errors": 
    [
        "Error: ENOENT, open '/u/echidna-staging/47aadb10-02fc-474a-9f1c-69b59f552b29//img/fingerprint.png'"
    ]
},
...
```

### PR DESCRIPTION
**Pull request by deniak**
Fix #119

### CODE REVIEW

File: lib/document-downloader.js [hunk 1]
```diff
@@ -38,6 +39,8 @@ DocumentDownloader.install = function (dest, content) {
 DocumentDownloader.installAll = function (destsContents) {
   // `e` is a tuple of [dest, content]
   return Promise.all(destsContents.toArray().map(function (e) {
+    var path = require('path').dirname(e[0]);
+    Mkdirp.sync(path);
```
Comment (astorije) [inline lib/document-downloader.js]: This sync version is fine by me.
I don't think it takes a lot longer to execute but I'd be curious to see the difference. Meanwhile it makes the code easier to read than a callback-based async version. When (if!) `mkdirp` [supports Promises](https://github.com/substack/node-mkdirp/issues/86), we can probably do something like:

```js
return Mkdirp(path).then(function ()
  return DocumentDownloader.install(e[0], e[1]);
});
```

File: package.json [hunk 2]
```diff
@@ -52,6 +52,7 @@
     "promise": "6.x.x",
     "request": "2.x.x",
     "specberus": "1.x.x",
+    "mkdirp": "0.x.x",
```
Comment (astorije) [inline package.json]: To match the other versions, this should be `0.5.x` instead.
Would that be a problem for you?

Comment (deniak) [inline package.json]: Done in a0d9f58beeecf4063ff14be07ef8dd1855a3197b

**Comment by astorije** [comment]
*A posteriori* :+1: :-)

**Pull request merged**